docs/rfc/rfc-018-bls-agg-exploration.md

```diff
@@ -0,0 +1,400 @@
+as state updates, the set of data written into the chain and received by the
+IBC module would be slightly smaller.
+
+## Discussion
+
+### What are the proposed benefits to aggregated signatures?
+
+#### Reduce Block Size
+
+At the moment, the commit size contains a 512 bit signature for each validator
+that voted for the block. For the cosmos hub, which has 175 validators in the
+validator set, this amounts to about 11kb per block. That is around 113 GB over
+the lifetime of the chain's 10118561 blocks. (Note, the hub has increased the
+number of validators in the set over time so the signature size over the history
+of the chain is likely less than that value). 
+
+Signature aggregation would only produce two signatures for the entire block.
+One for the yays and one for the nays. Each BLS aggregated signature is 48
+bytes, per the [IETF standard of BLS signatures][bls-ietf-ecdsa-compare].
+Over the lifetime of the same cosmos hub chain, that would amount to about 1
+GB, a savings of 112 GB. While that is a large factor of reduction it's worth
+bearing in mind that, at [GCP's cost][gcp-storage-pricing] of $.026 USD per GB,
+that is a total savings of around $2.50 per month.
+
+
+
+
+#### Reduce Signature Verification Creation and Verification Time
+
+From the [IETF draft standard on BLS Signatures][bls-ietf], BLS signatures can be
+created in 370 microseconds and verified in 2700 microseconds. Our current
+[Ed25519 implementation][voi-ed25519-perf] lists a 27.5 microsecond signature creation time
+and 5.19 milliseconds to perform batch verification on 128 signatures, which is
+slightly fewer than the 175 in the hub. blst, a popular implementation of BLS
+signature aggregation was benchmarked to perform verification on 100 signatures in
+1.5 milliseconds [when run locally][blst-verify-bench] on an 8 thread machine and
+pre-aggregated public keys.
+
+#### Reduce Light-Client Verification Time
+
+The light client aims to be a faster and lighter-weight way to verify that a 
+block was voted on by a Tendermint network. The light client fetches
+Tendermint block headers and commit signatures, performing public key
+verification to ensure that the associated validator set signed the block.
+Reducing the size of the commit signature would allow the light client to fetch
+block data more quickly.
+
+Additionally, the faster signature verification times of BLS signatures mean
+that light client verification would proceed more quickly.
+
+#### Reduce Gossip Bandwidth
+
+##### Vote Gossip
+
+It is possible to aggregate signatures during voting and not need to gossip all 
+*n* validator signatures to all other validators. Theoretically, subsets of
+the signatures could be aggregated during consensus to produce vote messages
+carrying aggregated signatures. Implementations of this would certainly
+increase the complexity of the gossip layer but may possibly reduce the total
+number of signatures required to be verified by each validator.
```

Comment:
```suggestion
It is possible to aggregate subsets of signatures during voting, so that the
network need not gossip all *n* validator signatures to all *n* validators.
Theoretically, subsets of the signatures could be aggregated during consensus
and vote messages could carry those aggregated signatures. Implementing this
would certainly increase the complexity of the gossip layer but could possibly
reduce the total number of signatures required to be verified by each validator.
```

For me the most damning part of this idea is dragging fundamental cryptographic key material into the gossip layer.